engine/CMakeLists.txt

```diff
@@ -1182,22 +1182,32 @@ ELSEIF (CMAKE_SYSTEM_NAME STREQUAL Linux)
 
 	# Lookup the LSB Data - this is known for Ubuntu, but also used by other distros too.
 	# and may help with distro-sub-selection
-	IF (EXISTS "/etc/lsb-release" )
-		MESSAGE("Found /etc/lsb-release")
-		SET(LSB_EXISTS "LSB Data IS available" ON)
-		# Ubuntu Version: /etc/lsb-release - DISTRIB_ID=Ubuntu, DISTRIB_CODENAME=<name>
-		FILE(STRINGS "/etc/lsb-release" LSB_DATA)
-		FOREACH(LSB_DATA_ENTRY IN LISTS LSB_DATA)
-			MESSAGE("Checking ${LSB_DATA_ENTRY} for match...")
-			IF ("${LSB_DATA_ENTRY}" MATCHES "^DISTRIB_ID=(.+)$")
-				SET(LSB_LINUX_DISTRIBUTION "${CMAKE_MATCH_1}")
-			ELSEIF ("${LSB_DATA_ENTRY}" MATCHES "^DISTRIB_CODENAME=(.+)$")
-				SET(LSB_LINUX_DISTRIBUTION_CODENAME "${CMAKE_MATCH_1}")
-			ENDIF ("${LSB_DATA_ENTRY}" MATCHES "^DISTRIB_ID=(.+)$")
-		ENDFOREACH(LSB_DATA_ENTRY)
-		MESSAGE("Found Linux Distribution: ${LSB_LINUX_DISTRIBUTION}")
-		MESSAGE("Found Linux Distribution Release Name: ${LSB_LINUX_DISTRIBUTION_CODENAME}")
-	ENDIF (EXISTS "/etc/lsb-release")
+    FIND_PROGRAM(LSB_RELEASE_EXEC lsb_release)
+    IF (LSB_RELEASE_EXEC)
+        EXECUTE_PROCESS(COMMAND ${LSB_RELEASE_EXEC} --codename --short
+            OUTPUT_VARIABLE LSB_LINUX_DISTRIBUTION_CODENAME
+            OUTPUT_STRIP_TRAILING_WHITESPACE)
+        SET(LSB_EXISTS "LSB Data IS available" ON)
+        MESSAGE("Found Linux Distribution Release Name: ${LSB_LINUX_DISTRIBUTION_CODENAME}")
+    ENDIF (LSB_RELEASE_EXEC)
+    MESSAGE("Found Linux Distribution Release Name: ${LSB_LINUX_DISTRIBUTION_CODENAME}")
```

Comment:
I've had mix results using `lsb_release` which is why I just read the `/etc/lsb-release` file directly; IIRC both are part of the LSB standard so either method is acceptable; the `lsb_release` binary should make it more resilient to changes in the format of `/etc/lsb-release` in theory.

Reply:
NOTE: I do like the conciseness of the change.

Reply:
I adapted this solution from https://stackoverflow.com/a/41592428/5067822 .

Reply:
It works on my machine (Ubuntu 20.04 focal); we can try it on others, and make sure it works there as well.